### PR TITLE
Folders/Files views {-||-} early impl...

### PR DESCRIPTION
**Pull request by dgt41**
#### Views Folders -- Files

Init code...

### CODE REVIEW

File: administrator/components/com_media/controllers/folders.php [hunk 1]
```diff
@@ -8,3 +8,16 @@
...
 
 defined('_JEXEC') or die;
+
+jimport('joomla.filesystem.file');
+jimport('joomla.filesystem.folder');
+
+/**
+ * Folder Media Controller
+ *
+ * @since  1.5
```
Comment (roland-d) [inline administrator/components/com_media/controllers/folders.php]: Version number is incorrect

File: administrator/components/com_media/models/folder.php [hunk 2]
```diff
@@ -21,4 +21,117 @@ class MediaModelFolder extends JModelLegacy
...
+
+	/**
+	 * Method to get model state variables
+	 *
+	 * @param   string $property Optional parameter name
+	 * @param   mixed  $default  Optional default value
+	 *
+	 * @return  object  The property where specified, the state object where omitted
+	 *
+	 * @since   1.5
```
Comment (roland-d) [inline administrator/components/com_media/models/folder.php]: Version number is incorrect

File: administrator/components/com_media/models/folder.php [hunk 3]
```diff
@@ -21,4 +21,117 @@ class MediaModelFolder extends JModelLegacy
...
+			$set = true;
+		}
+
+		return parent::getState($property, $default);
+	}
+
+	/**
+	 * Build browsable list of files
+	 *
+	 * @return  array
```
Comment (roland-d) [inline administrator/components/com_media/models/folder.php]: Please fix docbloc

File: administrator/components/com_media/models/folder.php [hunk 4]
```diff
@@ -21,4 +21,117 @@ class MediaModelFolder extends JModelLegacy
...
+//		$current = (string) $this->getState('folder');
+//		$currentFolder  = COM_MEDIA_BASE . ((strlen($current) > 0) ? '/' . $current : '');
+//
+//		return $currentFolder;
+//	}
+
+	/**
+	 * Return the files model
+	 *
+	 * @return MediaModelFiles
```
Comment (roland-d) [inline administrator/components/com_media/models/folder.php]: Please fix docbloc

File: administrator/components/com_media/models/folder.php [hunk 5]
```diff
@@ -21,4 +21,117 @@ class MediaModelFolder extends JModelLegacy
...
+	 */
+	protected function getFilesModel()
+	{
+		return new MediaModelFiles;
+	}
+
+	/**
+	 * Return the folder model
+	 *
+	 * @return MediaModelFolders
```
Comment (roland-d) [inline administrator/components/com_media/models/folder.php]: Please fix docbloc

File: administrator/components/com_media/models/folders.php [hunk 6]
```diff
@@ -41,7 +44,9 @@ public function getCurrentFolder()
 	/**
 	 * Set the current folder
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Please fix docbloc

File: administrator/components/com_media/models/folders.php [hunk 7]
```diff
@@ -51,54 +56,117 @@ public function setCurrentFolder($currentFolder)
 	}
 
 	/**
-	 * Build browsable list of files
+	 * Return the file model
+	 *
+	 * @return MediaModelFolder
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Please fix docbloc

File: administrator/components/com_media/models/folders.php [hunk 8]
```diff
@@ -51,54 +56,117 @@ public function setCurrentFolder($currentFolder)
...
+	{
+		return new MediaModelFolder;
+	}
+
+	/**
+	 * Get the folder tree
+	 *
+	 * @param   mixed  $base  Base folder | null for using base media folder
 	 *
 	 * @return  array
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Please fix docbloc

File: administrator/components/com_media/models/folders.php [hunk 9]
```diff
@@ -51,54 +56,117 @@ public function setCurrentFolder($currentFolder)
...
+	 *
+	 * @since   1.5
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Version number is incorrect

File: administrator/components/com_media/models/folders.php [hunk 10]
```diff
@@ -51,54 +56,117 @@ public function setCurrentFolder($currentFolder)
...
-	 * Return the file model
+	 * Method to get model state variables
 	 *
-	 * @return MediaModelFolder
+	 * @param   string  $property  Optional parameter name
+	 * @param   mixed   $default   Optional default value
+	 *
+	 * @return  object  The property where specified, the state object where omitted
+	 *
+	 * @since   1.5
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Version is incorrect

File: administrator/components/com_media/models/folders.php [hunk 11]
```diff
@@ -51,54 +56,117 @@ public function setCurrentFolder($currentFolder)
...
+			$parent = ($parent == '.') ? null : $parent;
+			$this->setState('parent', $parent);
+			$set = true;
+		}
+
+		return parent::getState($property, $default);
+	}
+
+	/**
+	 * Return the folder model
```
Comment (roland-d) [inline administrator/components/com_media/models/folders.php]: Please fix docbloc

File: administrator/components/com_media/views/folders/tmpl/default.php [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+					<div class="view">
+						<iframe class="thumbnail" src="index.php?option=com_media&amp;view=files&amp;tmpl=component&amp;folder=<?php echo "/images/";?>" id="folderframe" name="folderframe" width="100%" height="500px" marginwidth="0" marginheight="0" scrolling="auto"></iframe>
+					</div>
+					<?php echo JHtml::_('form.token'); ?>
+				</div>
+			</form>
+		</div>
+		<?php // Pre render all the bootstrap modals on the parent window
+
+		//echo JHtml::_(
```
Comment (roland-d) [inline administrator/components/com_media/views/folders/tmpl/default.php]: Should we remove the commented code?

File: administrator/components/com_media/views/folders/view.html.php [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+ * @copyright   Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * HTML View class for the Media component
+ *
+ * @since  1.0
```
Comment (roland-d) [inline administrator/components/com_media/views/folders/view.html.php]: Version is incorrect

**Comment by roland-d** [comment]
@dgt41 As a general remark, should we remove the commented out code?

**Comment by dgt41** [comment]
@roland-d This is the premature base that I will work on, so nothing here is actually ready or working as intended. I hope I will update this PR this weekend with the basic AJAX functionality and the layouts for each media type...

**Comment by yireo** [comment]
Could somebody submit this PR, so we can sync progress?

**Pull request merged**